TensorMath.lua

```diff
@@ -29,12 +29,14 @@ local unpack = unpack or table.unpack
 
 -- specific to CUDA
 local typenames = {'CudaByteTensor',
-                   'CudaCharTensor',
-                   'CudaShortTensor',
-                   'CudaIntTensor',
-                   'CudaLongTensor',
-                   'CudaTensor',
-                   'CudaDoubleTensor'}
+               'CudaCharTensor',
+               'CudaShortTensor',
+               'CudaIntTensor',
+               'CudaLongTensor',
+               'CudaTensor',
+               'CudaDoubleTensor',
+               'CudaHalfTensor'
```

Comment:
we cant flag it in the lua program, as this is essentially:
th TensorMath.lua >TensorMath.c

it is not aware of cutorch.

However, if you see, I've added #ifdef calls to hack around that.